### PR TITLE
backporting to CakePHP 2.x

### PR DESCRIPTION
**Pull request by bancer**
See https://github.com/cakephp/cakephp/pull/10985#issuecomment-320225098

@lorenzo You may want to pull to another branch though.

### CODE REVIEW

**Comment by lorenzo** [comment]
@bancer did it work for you?

**Comment by bancer** [comment]
Worked on a small test class (7 tests, 56 fixtures). I would need to test more in a bigger project on Monday.

**Comment by tersmitten** [comment]
Works for us :-) Timed in my local development machine (vagrant).

```
Tests: 820, Assertions: 7131, Fixtures: 107
```

**Before**
```
Time: 8.44 minutes, Memory: 99.79MB
```

**After**
```
Time: 6.52 minutes, Memory: 98.78MB
```

**Comment by tersmitten** [comment]
On our build server it is about 50 percent faster. 

4 min 23 sec versus 2 min 35 sec

**Comment by josegonzalez** [comment]
I changed the merge base to a 2.x branch for you, @lorenzo :)

File: Cake/TestSuite/Fixture/ChecksumTestFixture.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ */
+class ChecksumTestFixture extends CakeTestFixture {
+
+/**
+ * List of table hashes
+ *
+ * @var array
+ */
+	public static $_tableHashes = [];
```
Comment (bancer) [inline Cake/TestSuite/Fixture/ChecksumTestFixture.php:19]:
```
FILE: ...com/app/Plugin/Fixturize/Cake/TestSuite/Fixture/ChecksumTestFixture.php
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 19 | ERROR | Public member variable "_tableHashes" must not contain a leading
    |       | underscore
--------------------------------------------------------------------------------
```

Should that be protected or renamed to $tableHashes?

Comment (tersmitten) [reply Cake/TestSuite/Fixture/ChecksumTestFixture.php:19]: I changed it to be protected and that works just fine

**Comment by bancer** [comment]
The performance has really improved. We have a batch of all controller tests.

Without Fixturize:
```
Time: 19.54 minutes, Memory: 344.75MB

OK, but incomplete or skipped tests!
Tests: 948, Assertions: 3505, Incomplete: 56, Skipped: 9.
```

With Fixturize:
```
Time: 8.3 minutes, Memory: 346.50MB

OK, but incomplete or skipped tests!
Tests: 948, Assertions: 3505, Incomplete: 56, Skipped: 9.
```

In total we have approx. 180 fixtures but not all of them are used in every test class.

**Comment by lorenzo** [comment]
Thanks @bancer

**Pull request merged**